stack_test.go

```diff
@@ -0,0 +1,22 @@
+package errors
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestCaller(t *testing.T) {
+	s := NewStackTrace(0)
+	assert.NotEmpty(t, s.StackTrace)
+
+	assert.Equal(t, fmt.Sprintf("%s", s), "[stack_test.go testing.go asm_amd64.s]")
```

Comment:
Having a condition per architecture is a mess. In the end, what are the odds someone's going to run these tests using something that's not `amd64`?